runtime/engine/src/test/java/io/aklivity/zilla/runtime/engine/test/internal/binding/TestBindingFactory.java

```diff
@@ -246,9 +250,16 @@ private void onInitialBegin(
 
             if (catalogs != null)
             {
-                for (CatalogHandler catalog : catalogs)
+                for (CatalogHandler handler : catalogs)
                 {
-                    catalog.resolve(0);
+                    if (catalog.subject != null && catalog.version != null)
+                    {
+                        handler.resolve(catalog.subject, catalog.version);
+                    }
+                    else
+                    {
+                        handler.resolve(catalog.id);
+                    }
```

Comment:
This is excellent. As mentioned above, we may need to add support for verifying the expected schema here, then reject the initial begin with a reset if the schema doesn't match expectations, as that will cause the IT to fail.

The expected schemas may be a separate test binding option, independent of cataloged so as not to interfere with that generic definition used by other bindings that don't need to validate expectations.